src/test/scala/io/iohk/ethereum/blockchain/sync/regular/BlockFetcherSpec.scala

```diff
@@ -124,6 +124,78 @@ class BlockFetcherSpec extends TestKit(ActorSystem("BlockFetcherSpec_System")) w
       peersClient.expectMsgClass(classOf[BlacklistPeer])
       peersClient.expectMsgPF() { case PeersClient.Request(msg, _, _) if msg == firstGetBlockHeadersRequest => () }
     }
+
+    "should ensure blocks passed to importer are always forming chain" in new TestSetup {
```

Comment:
Could this test be more similar to the ones above? I mean, instead of invalidation, here we are mining block in the middle...I think this sequence should be enough:

- fetcher request headers
-  we mine a new block (block 1 from a fork)
- fetcher should receive those blocks bodies
- the resulted chain shouldn't have gaps.

Reply:
I'm not sure. Whole issue was that if block was mined between 2 headers requests, gap was created because not all of headers were added to queue. Scenario you described seem to be more appropriate to check wrt received block bodies validation.

In scenario you described we still would need to either prepare fetcher somehow with some branch - which essentially would evolve it to the same as it is now.

Reply:
ok, no requested changes for now then! =)